internal/transport/transport_test.go

```diff
@@ -1663,6 +1663,82 @@ func (s) TestReadGivesSameErrorAfterAnyErrorOccurs(t *testing.T) {
 	}
 }
 
+func (s) TestServerWithClientSendingWrongMethod(t *testing.T) {
```

Comment:
A function level comment describing what the test is doing would be useful.

Reply:
Added.